src/v/cluster/controller_backend.cc

```diff
@@ -581,20 +590,37 @@ controller_backend::calculate_learner_initial_offset(
         return std::nullopt;
     }
 
-    const auto max_removable_local_log_offset
-      = p->max_removable_local_log_offset();
+    auto max_removable_local_log_offset = p->max_removable_local_log_offset();
+    auto last_uploaded = p->archival_meta_stm()->manifest().get_last_offset();
+
     /**
      * Last offset uploaded to the cloud is target learner retention upper
      * bound. We can not start retention recover from the point which is not yet
      * uploaded to Cloud Storage.
+     *
+     * In general max_removable_local_log_offset should not exceed
+     * last_uploaded, but can if, for example, archival is disabled or paused.
      */
+
+    if (max_removable_local_log_offset > last_uploaded) {
+        vlog(
+          clusterlog.info,
+          "[{}] max_removable_local_log_offset {} exceeds last uploaded to "
+          "cloud {}, clamping to {}",
+          p->ntp(),
+          max_removable_local_log_offset,
+          last_uploaded,
+          last_uploaded);
+        max_removable_local_log_offset = last_uploaded;
```

Comment:
this should be `std::min(stm.get_last_offset(), stm.get_last_clean_at());`

Reply:
also, the TS is disabled maybe it makes more sense to just copy the log as if there is no TS to begin with